copy.go

```diff
@@ -235,3 +238,16 @@ func fclose(f *os.File, reported *error) {
 		*reported = err
 	}
 }
+
+// onError lets caller to handle errors
+// occured when copying a file.
+func onError(err error, opt Options) error {
+	if err == nil {
```

Comment:
I don't think we need this `err == nil` check. Whether nil or not, we should pass it to following steps.

Reply:
this was to avoid reporting onError to a hook. i updated it to report anyways.